packer/rpc/post_processor.go

```diff
@@ -99,7 +99,6 @@ func (p *PostProcessorServer) PostProcess(streamId uint32, reply *PostProcessorP
 	if err != nil {
 		return NewBasicError(err)
 	}
-	defer client.Close()
```

Comment:
Found it: here in this func sometimes PostProcess line 109 returns the same Artifact from `client.Artifact()`; but then if we close `client` then no calls to `Artifact` will work.

Reply:
The this is that it works in some cases and I don't know why...

Reply:
Did you ever figure this out?

Reply:
By removing this Close() do we leave child processes running after Packer exits?

Reply:
ah I see the logic below. Asked too soon.

Reply:
Yes it stays running forever; hopefully it's just in a waiting state and does nothing and this happens only for that case.

Another good reason to move over to hashicorp/go-plugin 🙂